src/includes/sourcemaps/generate/javascript.mdx

```diff
@@ -0,0 +1,166 @@
+
+</Note>
+
+### Advanced Usage
+
+If you prefer to upload source maps manually, configure Webpack to output source maps:
+
+<PlatformContent includePath="configuration/sourcemaps/tools/webpack/advanced-usage" />
+
+If you use [SourceMapDevToolPlugin](https://webpack.js.org/plugins/source-map-dev-tool-plugin) for more fine-grained control of source map generation, turn off `noSources` so Sentry can display proper source code context in event stack traces.
+
+Additionally, the Webpack plugin will automatically set `window.SENTRY_RELEASE`, so your `Sentry.init` call will not need to be updated.
+
+## Rollup
+
+You can configure Rollup to generate source maps:
+
+```javascript {filename:rollup.config.js}
+export default {
+  entry: "./src/app.js",
+  output: {
+    file: "bundle.js",
+    format: "cjs",
+    sourceMap: true,
+  },
+};
+```
+
+<PlatformSection notSupported={["javascript.nextjs"]}>
+
+## SystemJS
+
+SystemJS is the default module loader for Angular 2 projects. The [SystemJS build tool](https://github.com/systemjs/builder) can be used to bundle, transpile, and minify source code for use in production environments, and can be configured to output source maps.
+
+```javascript
+builder.bundle("src/app.js", "dist/app.min.js", {
+  minify: true,
+  sourceMaps: true,
+  sourceMapContents: true,
+});
+```
+
+<Note>
+
+This example configuration will inline your original, un-transformed source code into the generated source map file. Sentry requires both source map(s) **and** your original source files to perform reverse transformations. If you choose NOT to inline your source files, you must make those source files available to Sentry _in addition_ to your source maps (see below).
+
+</Note>
+
+</PlatformSection>
+
+## TypeScript
+
+The TypeScript compiler can output source maps. Configure the `sourceRoot` property to `/` to strip the build path prefix from generated source code references. This allows Sentry to match source files relative to your source root folder:
+
+```javascript {filename:tsconfig.json}
+{
+  "compilerOptions": {
+    "sourceMap": true,
+    "inlineSources": true,
+    "sourceRoot": "/"
+  }
+}
+```
+
+<PlatformSection notSupported={["javascript.nextjs"]}>
+
+## UglifyJS
+
+UglifyJS is a popular tool for minifying your source code for production. It can dramatically reduce the size of your files by eliminating whitespace, rewriting variable names, removing dead code branches, and more.
+
+<Note>
+
+We strongly encourage you to use a higher level bundler (or transpiler) as UglifyJS configuration can get quite complicated to achieve the desired results.
+
+</Note>
+
+If you are using UglifyJS to minify your source code, the following command will additionally generate a source map that maps the minified code back to the original source:
+
+```bash
+uglifyjs app.js \
+  -o app.min.js.map \
+  --source-map url=app.min.js.map,includeSources
+```
+
+</PlatformSection>
```

Comment:
Got rid of the platform section (since nextjs shouldn't have this page at all) and took out the blurb about Uglify. (See notes about both of these in previous comments.)

NB: Also see note above re: backticks.

```suggestion
## UglifyJS

<Note>

We strongly encourage you to use a higher-level bundler (or transpiler), as UglifyJS configuration can get quite complicated and it is difficult to achieve the desired results.

</Note>

[UglifyJS](https://github.com/mishoo/UglifyJS) can be configured to output source maps, which you can then [upload using `sentry-cli`](/product/cli/releases/#sentry-cli-sourcemaps):

```bash
uglifyjs app.js \
  -o app.min.js.map \
  --source-map url=app.min.js.map,includeSources
// TODO: uncomment the backticks below
// ```
```